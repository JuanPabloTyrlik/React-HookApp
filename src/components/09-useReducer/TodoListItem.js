import React from 'react';
import PropTypes from 'prop-types';

const TodoListItem = ({ todo, i, handleDelete, handleToggle }) => {
    return (
        <li key={todo.id} className="list-group-item">
            <p
                className={`${todo.done && 'complete'}`}
                onClick={() => handleToggle(todo)}
            >
                {i + 1}. {todo.desc}
            </p>
            <button
                onClick={() => handleDelete(todo)}
                className="btn btn-danger"
            >
                Borrar
            </button>
        </li>
    );
};

TodoListItem.propTypes = {
    todo: PropTypes.object.isRequired,
    i: PropTypes.number.isRequired,
    handleDelete: PropTypes.func.isRequired,
    handleToggle: PropTypes.func.isRequired,
};

export default TodoListItem;
