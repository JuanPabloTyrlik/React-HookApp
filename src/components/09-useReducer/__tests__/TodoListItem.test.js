import { shallow } from 'enzyme';
import React from 'react';
import TodoListItem from '../TodoListItem';

describe('Tests on TodoListItem', () => {
    const demoTodos = [
        {
            id: 1,
            desc: 'Aprender React',
            done: true,
        },
        {
            id: 2,
            desc: 'Aprender Mongo',
            done: false,
        },
    ];
    const handleDelete = jest.fn();
    const handleToggle = jest.fn();
    const wrapper = shallow(
        <TodoListItem
            key={demoTodos[0].id}
            todo={demoTodos[0]}
            i={demoTodos[0].id}
            handleToggle={handleToggle}
            handleDelete={handleDelete}
        />
    );

    test('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('should call handleDelete', () => {
        wrapper.find('button').simulate('click');
        expect(handleDelete).toHaveBeenCalled();
        expect(handleDelete).toHaveBeenCalledWith(demoTodos[0]);
    });
    test('should call handleToggle', () => {
        wrapper.find('p').simulate('click');
        expect(handleToggle).toHaveBeenCalled();
        expect(handleToggle).toHaveBeenCalledWith(demoTodos[0]);
    });
    test('should render text correctly', () => {
        expect(wrapper.find('p').text()).toMatch(`2. Aprender React`);
    });
    test('should cross out item', () => {
        expect(wrapper.find('p').hasClass('complete')).toBe(true);
    });
});
