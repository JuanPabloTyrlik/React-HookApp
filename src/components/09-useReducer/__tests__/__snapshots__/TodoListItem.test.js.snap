// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on TodoListItem should render correctly 1`] = `
<li
  className="list-group-item"
  key="1"
>
  <p
    className="complete"
    onClick={[Function]}
  >
    2
    . 
    Aprender React
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Borrar
  </button>
</li>
`;
